src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -619,18 +697,44 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
   }
 
   function performWithPriority(priorityLevel : PriorityLevel, fn : Function) {
-    const previousDefaultPriority = defaultPriority;
-    defaultPriority = priorityLevel;
+    const previousPriorityContext = priorityContext;
+    priorityContext = priorityLevel;
     try {
       fn();
     } finally {
-      defaultPriority = previousDefaultPriority;
+      priorityContext = previousPriorityContext;
+    }
+  }
+
+  function batchedUpdates<A>(fn : () => A) : A {
+    const prev = shouldBatchUpdates;
+    shouldBatchUpdates = true;
+    try {
+      return fn();
+    } finally {
+      shouldBatchUpdates = prev;
+      // If we've exited the batch, perform any scheduled sync work
+      if (!shouldBatchUpdates) {
+        performSynchronousWork();
```

Comment:
Should this be performed if an error is thrown? Currently it would be since it is in `finally`. cc @gaearon